### PR TITLE
Add `private_key_jwt` client authentication method

### PR DESCRIPTION
**Pull request by lbeaufort**
Issue https://github.com/fecgov/fecfile-web-api/issues/90

### Testing
**Run automated tests**
```
$ virtualenv -p /path/to/bin/python3.8 venv
$ source venv
(venv) $ pip install -r requirements/requirements_dev.txt
(venv) $ DJANGO_SETTINGS_MODULE=tests.settings django-admin.py test
```
**Test with demo app**
https://login-demo.app.cloud.gov/simpleapp/

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -37,6 +37,7 @@
     'josepy',
     'requests',
     'cryptography',
+    'pyjwt',
```
Comment (lbeaufort) [inline setup.py:40]: Do we need to pin this version?

**Review by dheitzer — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (dheitzer) [reply setup.py:40]: will investigate as part api #109

File: mozilla_django_oidc/utils.py [hunk 2]
```diff
@@ -97,3 +100,13 @@ def add_state_and_nonce_to_session(request, state, params):
...
     }
+
+
+def add_state_to_cookie(response, state):
+    """
+    Adds state to cookie for logout
+    """
+    # TODO: add test cookie. If no cookie, end all sessions
+    response.set_signed_cookie('oidc_state', state)
+
```
Comment (lbeaufort) [inline mozilla_django_oidc/utils.py:111]: Need a plan if user has cookies turned off. Required? Login.gov requires the `state` value for logout.

Comment (lbeaufort) [reply mozilla_django_oidc/utils.py:111]: But also if the front end needs to save the JWT for API calls, it needs to be saved in the browser.

Multiple tabs/devices are also a use case.

**Pull request merged**